sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileIndex.scala

```diff
@@ -51,12 +51,8 @@ trait FileIndex {
    *                         files where these predicates are guaranteed to evaluate to `true`.
    *                         Thus, these filters will not need to be evaluated again on the
    *                         returned data.
-   * @param dataFilters Filters that can be applied on non-partitioned columns. The implementation
-   *                    does not need to guarantee these filters are applied, i.e. the execution
-   *                    engine will ensure these filters are still applied on the returned files.
    */
-  def listFiles(
-      partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Seq[PartitionDirectory]
+  def listPartitionData(partitionFilters: Seq[Expression]): Seq[PartitionDirectory]
```

Comment:
Since dataFilters is not used in all sub classes of FileIndex, so just remove the dataFilters parameter here.

Reply:
@fuwhu I would suggest keep the current method with parameter `dataFilters`.
See the discussions in https://github.com/apache/spark/pull/27157
Also, the rename is not related to the proposal of this PR, right?

Reply:
yea, just changed it since i think this method return sequence of PartitionDirectory objects, which is not actually to list file.
Will change it back to keep PR proposal clear.

Reply:
@gengliangwang  Removed the conf MAX_PARTITION_NUMBER_FOR_STATS_CALCULATION_VIA_FS in this PR, since it is not needed per discussion in #27129 .